app/controllers/posts_controller.rb

```diff
@@ -7,4 +10,33 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
+
+  def find_post
+    @post = Post.find(params[:id])
+  end
+
+  def current_user
+    @current_user = User.find(session[:user_id]) if session[:user_id]
+  end
```

Comment:
- [ ] please use the application_controller for this method so it is available in the controllers.

Reply:
Oops, now I've got it. I was wondering why my code was not working properly. I looked at others' code and even copied some, thinking I was writing it wrong all the time. Thank you for your review :)